### DISCUSSION TITLE
How to get readings from multi-ranger deck?

### DISCUSSION THREAD

**Discussion by sabdi1** [USC-ACTLab/crazyswarm#365] [Q&A]
I have a flow deck and multi-ranger deck on my crazyflie. I am able to retrieve pose information with help of flow deck. How can I get the readings of multi-ranger deck?

**Comment by sabdi1** [comment]
Hello @whoenig , are you aware if the capability of retrieving measurements from the multi-ranger has been implemented into Crazyswarm?

**Comment by whoenig** [comment]
There is no need for any special support. You can use the generic logging functionality with the variables defined at https://github.com/bitcraze/crazyflie-firmware/blob/a70c357a7c03b84a0caebe6d31a5fb0ccc40ff2b/src/modules/src/range.c#L59-L66. You can add those variables at https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L12-L14 and enable logging at https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/hover_swarm.launch#L48.
